services/venice-router/src/main/java/com/linkedin/venice/router/api/routing/helix/HelixGroupLeastLoadedStrategy.java

```diff
@@ -60,10 +61,21 @@ public int selectGroup(long requestId, int groupCount) {
       }
       for (int i = 0; i < groupCount; ++i) {
         int currentGroup = (i + startGroupId) % groupCount;
-        long currentGroupCounter = counters[currentGroup];
+        int currentGroupCounter = counters[currentGroup];
```

Comment:
instead of synchronized can we use `CopyOnWriteArrayList` for counters array?

Reply:
Counter will be updated per request, so it means the update frequency is fairly high, and I think `CopyOnWriteArrayList` is more suitable for less-update use cases and the logic being executed here is very fast, `synchronized` won't introduce much overhead.